locations/spiders/barlouie.py

```diff
@@ -0,0 +1,48 @@
+import json
+
+import scrapy
+
+from locations.items import GeojsonPointItem
+
+
+class BarlouieSpider(scrapy.Spider):
+    name = "barlouie"
+    allowed_domains = ["www.barlouie.com"]
+    start_urls = ["https://www.barlouie.com/locations"]
+    item_attributes = {"brand": "Barlouie", "brand_wikidata": "Q16935493"}
+
+    def parse(self, response):
+        for link in response.xpath("//a[contains(@href, 'locations')]/@href").getall():
+            yield scrapy.Request(
+                "https://www.barlouie.com" + link, self.parse_each_website
+            )
+
+    def parse_each_website(self, response):
+        for ldjson in response.xpath(
+            '//script[@type="application/ld+json"]/text()'
+        ).extract():
+            sanitized_data = self.sanitize_json(ldjson)
+            data = json.loads(sanitized_data)
+            if data["@type"] == "Restaurant":
+                yield self.parse_store(response, data)
+
+    def sanitize_json(self, data):
```

Comment:
sanitisation was needed because data was not in json format.
```
"openingHours": [
    "Mo-Fr 11:00-02:00","Sa-Su 10:00-02:00",
    "TimeZoneOffsetSeconds": 0,
    ]
    }
```
`"TimeZoneOffsetSeconds": 0,` should have been in curly braces as it is a json object and not a string. So removed this entire entry.

Reply:
It's a pain, but OK.

Maybe wrap it with an `if` so hopefully the spider continues to work after they fix the website.

Reply:
Wrapped in try-catch block.